src/library_syscall.js

```diff
@@ -227,7 +227,7 @@ var SyscallsLibrary = {
   },
   __syscall5: function(which, varargs) { // open
     var pathname = SYSCALLS.getStr(), flags = SYSCALLS.get(), mode = SYSCALLS.get() // optional TODO
-    var stream = FS.open(pathname, flags, mode);
+    var stream = FS.open(pathname, flags | {{{ cDefine('O_LARGEFILE') }}}, mode);
```

Comment:
Moved the flag back to libc in https://github.com/kripken/emscripten/pull/4813/commits/2c36bf0168e9f3b0138e8ac5b5b099f668bfab68

That required some shenaniganry to get around cyclical macro expansions, but overall I think that makes the change clearer.